### PR TITLE
Data lake support

### PR DESCRIPTION
**Pull request by mclacore**
* Added HNS (hierarchical namespace) to the export storage account, which is required per [docs](https://learn.microsoft.com/en-us/azure/healthcare-apis/fhir/export-data).
* Removed network ACLs from the storage accounts as adding additional containers to the storage account from my IP address was getting blocked. The storage accounts aren't exposed to security risks by removing network ACLs, they still require an access key or RBAC to access.

https://linear.app/massdriver/issue/ORC-181/azure-network-iam-check

### CODE REVIEW

File: src/log.tf [hunk 1]
```diff
@@ -24,10 +24,11 @@ resource "azurerm_storage_account" "diag-log" {
     type = "SystemAssigned"
   }
 
-  network_rules {
-    default_action = "Deny"
-    bypass         = ["AzureServices", "Logging"]
-  }
+  # network_rules {
```
Comment (WillBeebe) [inline src/log.tf:27]: We want to keep these

File: src/export.tf [hunk 2]
```diff
@@ -7,13 +7,15 @@ resource "azurerm_storage_account" "export" {
...
   enable_https_traffic_only = true
+  is_hns_enabled            = true
   min_tls_version           = "TLS1_2"
   tags                      = var.md_metadata.default_tags
 
-  network_rules {
-    default_action = "Deny"
-    bypass         = ["AzureServices", "Logging"]
-  }
+  # network_rules {
```
Comment (WillBeebe) [inline src/export.tf:14]: We want to keep these

File: src/log.tf (hunk 1 above)
Comment (mclacore) [reply src/log.tf]: In the description of the PR I explained why I removed them

**Comment by WillBeebe** [comment]
I could be wrong, FHIR might be different that app-service, function-app, etc but I've had luck adding this to the network rules.

https://github.com/massdriver-cloud/terraform-modules/blob/main/massdriver-application-azure-function-app/storage.tf#L24

Please complete AKS before coming back to this and please keep the tickets in Linear updated. I'd prefer if each person had their own ticket, and to not share tickets. Also, we have one for this specifically.

https://linear.app/massdriver/issue/ORC-185/make-sure-managed-identity-is-turned-on-for-fhir

I've marked it as TODO and assigned it to you. 👍

File: src/_artifacts.tf [hunk 3]
```diff
@@ -12,6 +12,18 @@ resource "massdriver_artifact" "fhir" {
           authority = azurerm_healthcare_fhir_service.main.authentication[0].authority
           audience  = azurerm_healthcare_fhir_service.main.authentication[0].audience
         }
+        security = {
```
Comment (WillBeebe) [inline src/_artifacts.tf:15]: adding RBAC, nice 👍

File: massdriver.yaml [hunk 4]
```diff
@@ -119,24 +119,43 @@ params:
         - allowed_origins
       properties:
         region:
-          # List of available regions for the resource type is southcentralus,northeurope,westeurope,eastus,eastus2,
-          # australiaeast,uksouth,westus2,canadacentral,switzerlandnorth,westus3,centralindia,southeastasia,koreacentral,
-          # swedencentral.
+          # This is the full list of regions for Azure FHIR Service. The list of Azure regions in the MD Azure region widget
```
Comment (WillBeebe) [inline massdriver.yaml:122]: @xpositivityx should we backlog a ticket to support this in Sat?

**Review by xpositivityx — APPROVED** [review]

File: src/main.tf [hunk 5]
```diff
@@ -35,7 +36,7 @@ resource "azurerm_healthcare_fhir_service" "main" {
     # authority is used for token validation
     authority = "https://login.microsoftonline.com/${var.azure_service_principal.data.tenant_id}"
     # audience identifies intended recipients of the token
-    audience = "https://${var.md_metadata.name_prefix}.fhir.azurehealthcareapis.com"
+    audience = "https://${azurerm_healthcare_workspace.main.name}-${var.md_metadata.name_prefix}.fhir.azurehealthcareapis.com"
```
Comment (WillBeebe) [inline src/main.tf:39]: Adding a prefix here reminds me of the DNS for app-service and function-app. 👍 

This is interesting though. The authority is active directory for this tenant, and the audience is important. I wonder if this has any influence on RBAC for this service.

**Review by WillBeebe — COMMENTED** [review]
I don't think I have the context on Datalake Support and it looks like the linked ticket is for something else. No worries, just a heads up. These changes are looking good though, I'm liking the RBAC additions and comments. Left a couple of comments / questions.

File: src/main.tf (hunk 5 above)
Comment (mclacore) [reply src/main.tf:39]: I'll find out and let you know!

**Comment by mclacore** [comment]
Taking data lake and moving it to it's own bundle.

**Pull request closed without merging**